UnityProject/Assets/Scripts/Core/UniversalObjectPhysics.cs

```diff
@@ -1711,61 +1677,119 @@ public void FlyingUpdateMe()
 				AppliedFriction(DEFAULT_Friction);
 			}
 		}
+	}
 
+	private void ProcessCollisionDetection(ref Vector3 position, ref Vector3 newPosition)
+	{
+		if (Collider != null) Collider.enabled = false;
+		var hit = MatrixManager.Linecast((position - (NewtonianMovement.normalized * 0.2f).To3()),
+			LayerTypeSelection.Walls | LayerTypeSelection.Grills | LayerTypeSelection.Windows,
+			defaultInteractionLayerMask, newPosition);
+		if (hit.ItHit)
+		{
+			OnImpact?.Invoke(this, NewtonianMovement);
+			NewtonianMovement -= 2 * (NewtonianMovement * hit.Normal) * hit.Normal;
+			var offset = (0.1f * hit.Normal);
+			newPosition = hit.HitWorld + offset.To3();
+			NewtonianMovement *= ObjectBouncyness;
+			spinMagnitude *= -1;
+			if (hit.CollisionHit.GameObject != null)
+			{
+				Hits.Add(hit.CollisionHit.GameObject.GetUniversalObjectPhysics());
+			}
+		}
+		if (Collider != null) Collider.enabled = true;
+	}
 
-		var position = this.transform.position;
+	private void ProcessThingsToHit()
+	{
+		if (Hits.Count > 0)
+		{
+			OnImpact?.Invoke(this, NewtonianMovement);
+		}
+		if (attributes.Component is not ItemAttributesV2 iav2) return;
+		foreach (var hit in Hits)
+		{
+			//TODO: DamageTile( goal,Matrix.Matrix.TilemapsDamage);
+			if (hit == null) continue;
+			if (NewtonianMovement.magnitude < iav2.ThrowSpeed * 0.25f) continue;
+			var damage = (iav2.ServerThrowDamage);
+			if (hit.TryGetComponent<Integrity>(out var integrity) && isServer)
+			{
+				integrity.ApplyDamage(damage, AttackType.Melee, iav2.ServerDamageType);
+			}
+			//TODO: Catching
+			//(Max): Catching what?
```

Comment:
In ss13 if you're holding the throw button, you can Catch items being thrown at you

Reply:
oh, you could have explained this further but sure ig.

Reply:
updated the comment to reflect that